### DISCUSSION TITLE
How do I set maxAge on session (so that login is remembered after browser is closed)?

### DISCUSSION THREAD

**Discussion by PiotrJander** [sergiodxa/remix-auth#197] [Q&A] [answered]
In a typical login form, there is a "remember" checkbox. If set, session cookies will have `maxAge` set so that they persist after the browser is closed.

In the Indie stack Remix starter, this is implemented like this:

```
export async function createUserSession({
  request,
  userId,
  remember,
  redirectTo,
}: {
  request: Request;
  userId: string;
  remember: boolean;
  redirectTo: string;
}) {
  const session = await getSession(request);
  session.set(USER_SESSION_KEY, userId);
  return redirect(redirectTo, {
    headers: {
      "Set-Cookie": await sessionStorage.commitSession(session, {
        maxAge: remember
          ? 60 * 60 * 24 * 7 // 7 days
          : undefined,
      }),
    },
  });
}
```

How do I achieve this with `remix-auth`?

Thanks!

**Comment by sergiodxa** [comment, marked as answer]
If you don't define a successRedirect when calling `authenticate` method it will give you back the user data and you'll have to commit the session yourself, at that point you can change the maxAge.

Check https://github.com/sergiodxa/remix-auth#custom-redirect-url-based-on-the-user for how this may work, the example is to change the redirect but the idea is mostly the same, just change where you commit the session.
